File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int answer;
    public int solution(int[] numbers, int target) {
        answer = 0;
        
        recur(0, 0, numbers, target);
        return answer;
    }
    public void recur(int idx,int num, int[] numbers, int target) {
        if (idx == numbers.length) {
            if (num == target) {
                answer++;
            }
            return;
            
        }
        recur(idx + 1, num - numbers[idx], numbers, target);
        recur(idx + 1, num + numbers[idx], numbers, target);
        
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <bits/stdc++.h>

using namespace std;

int ret;

void dfs(vector<int> numbers, int target, int idx, int now_val) {
    if (idx == numbers.size()) {
        if (target == now_val) {
            ret++;
        }
        return;
    }

    

    dfs(numbers, target, idx + 1, now_val + numbers[idx]);
    dfs(numbers, target, idx + 1, now_val - numbers[idx]);
}

int solution(vector<int> numbers, int target) {

    dfs(numbers, target, 0, 0);

    return ret;
}